internal/handlers/pg/pgdb/collections.go

```diff
@@ -144,8 +144,13 @@ func CreateCollection(ctx context.Context, tx pgx.Tx, db, collection string) err
 // If needed, it creates both database and collection.
 //
 // True is returned if collection was created.
-func CreateCollectionIfNotExist(ctx context.Context, tx pgx.Tx, db, collection string) (bool, error) {
-	exists, err := CollectionExists(ctx, tx, db, collection)
+func CreateCollectionIfNotExist(ctx context.Context, pgPool *Pool, db, collection string) (bool, error) {
```

Comment:
This probably is outside the scope of this task, just noting that even if collection did exist when this func was called, nothing is preventing from collection to be deleted after calling this func.

Reply:
Yeah, things would be much easier if we could have collection creation and inserting in the same transaction

Reply:
As discussed in the call, an alternative approach could be using the same transaction and retrying it. The problem with it is with ordered inserts there are some errors that could be retried (from db and collection creation) and errors that couldn't (from inserts themselves). With such approach, we need to find a way to handle those retries correctly...

@noisersup let's wait for more review, but personally I'd vote for merging the current solutions with multiple transactions.